plugins/destination/mysql/client/types.go

```diff
@@ -2,63 +2,66 @@ package client
-	case schema.TypeString,
-		schema.TypeStringArray,
-		schema.TypeUUIDArray,
-		schema.TypeCIDRArray,
-		schema.TypeMacAddrArray,
-		schema.TypeInetArray,
-		schema.TypeIntArray:
-		return "text"
-	case schema.TypeJSON:
-		return "json"
-	default:
-		panic("unknown type " + t.String())
+func mySQLTypeToArrowType(tableName string, columnName string, sqlType string) (arrow.DataType, error) {
+	if strings.HasPrefix(sqlType, "datetime") {
+		// MySQL permits up to microseconds (6 digits) precision
+		return arrow.FixedWidthTypes.Timestamp_us, nil
 	}
-}
-
-func SchemaType(tableName string, columnName string, sqlType string) (schema.ValueType, error) {
-	sqlTypeToSchemaType := map[string]schema.ValueType{
-		"bool":          schema.TypeBool,
-		"tinyint(1)":    schema.TypeBool,
-		"bigint":        schema.TypeInt,
-		"bigint(20)":    schema.TypeInt,
-		"float":         schema.TypeFloat,
-		"binary(16)":    schema.TypeUUID,
-		"blob":          schema.TypeByteArray,
-		"datetime":      schema.TypeTimestamp,
-		"nvarchar(255)": schema.TypeInet,
-		"varchar(255)":  schema.TypeInet,
-		"text":          schema.TypeString,
-		"json":          schema.TypeJSON,
+	sqlTypeToSchemaType := map[string]arrow.DataType{
+		"bool":          arrow.FixedWidthTypes.Boolean,
+		"tinyint(1)":    arrow.FixedWidthTypes.Boolean,
+		"bigint":        arrow.PrimitiveTypes.Int64,
+		"bigint(20)":    arrow.PrimitiveTypes.Int64,
+		"float":         arrow.PrimitiveTypes.Float64,
+		"binary(16)":    types.ExtensionTypes.UUID,
+		"blob":          arrow.BinaryTypes.LargeBinary,
+		"nvarchar(255)": types.ExtensionTypes.Inet,
+		"varchar(255)":  types.ExtensionTypes.Inet,
+		"text":          arrow.BinaryTypes.LargeString,
+		"json":          types.ExtensionTypes.JSON,
 	}
 
 	if v, ok := sqlTypeToSchemaType[sqlType]; ok {
 		return v, nil
 	}
 
-	return schema.TypeInvalid, fmt.Errorf("got unknown MySQL type %q for column %q of table %q while trying to convert it to CloudQuery internal schema type. Supported MySQL types are %q", sqlType, columnName, tableName, maps.Keys(sqlTypeToSchemaType))
+	supportedTypes := maps.Keys(sqlTypeToSchemaType)
+	supportedTypes = append(supportedTypes, "datetime")
+	sort.Strings(supportedTypes)
+	return nil, fmt.Errorf("got unknown MySQL type %q for column %q of table %q while trying to convert it to CloudQuery internal schema type. Supported MySQL types are %q", sqlType, columnName, tableName, supportedTypes)
+}
+
+func arrowTypeToMySqlStr(t arrow.DataType) string {
+	switch t.(type) {
+	case *arrow.BooleanType:
+		return "bool"
+	case *arrow.Int8Type, *arrow.Uint8Type, *arrow.Int16Type, *arrow.Uint16Type, *arrow.Int32Type, *arrow.Uint32Type, *arrow.Int64Type, *arrow.Uint64Type:
+		return "bigint"
+	case *arrow.Float16Type, *arrow.Float32Type, *arrow.Float64Type:
+		return "float"
+	case *arrow.StringType, *arrow.LargeStringType:
+		return "text"
+	case *arrow.BinaryType, *arrow.LargeBinaryType:
+		return "blob"
+	case *types.UUIDType:
+		return "binary(16)"
```

Comment:
Let's use string I think.

Reply:
There are a few options for UUID:
 - `binary(16)` is the most efficient
 - `VARCHAR(36)` or VARCHAR(255) which will show nicely in queries
 - `text` which is less efficient
 
 The thing is that when we query the DB during migration we can't 100% say that `VARCHAR(36)` or `VARCHAR(255)` is UUID unless we try to parse the data.
 
 Assuming `binary(16)` is UUID is safer (but also not 100%). So the choice for `binary(16)` was kind of a tradeoff to avoid using `text`.
 
 We have a similar issue with `bool` which is an alias for `tinyint(1)`, so we assume `tinyint(1)` is boolean